cupy_backends/cuda/api/driver.pyx

```diff
@@ -15,89 +15,23 @@ cimport cython  # NOQA
-                      unsigned int numOptions, CUjit_option* options, void**
-                      optionValues)
-    int cuLinkComplete(LinkState state, void** cubinOut, size_t* sizeOut)
-    int cuLinkDestroy(LinkState state)
-    int cuModuleLoad(Module* module, char* fname)
-    int cuModuleLoadData(Module* module, void* image)
-    int cuModuleUnload(Module hmod)
-    int cuModuleGetFunction(Function* hfunc, Module hmod,
-                            char* name)
-    int cuModuleGetGlobal(Deviceptr* dptr, size_t* bytes, Module hmod,
-                          char* name)
-    int cuModuleGetTexRef(TexRef* pTexRef, Module hmod, const char* name)
-    int cuLaunchKernel(
-        Function f, unsigned int gridDimX, unsigned int gridDimY,
-        unsigned int gridDimZ, unsigned int blockDimX,
-        unsigned int blockDimY, unsigned int blockDimZ,
-        unsigned int sharedMemBytes, Stream hStream,
-        void** kernelParams, void** extra)
-
-    int cuLaunchCooperativeKernel(
-        Function f, unsigned int gridDimX, unsigned int gridDimY,
-        unsigned int gridDimZ, unsigned int blockDimX,
-        unsigned int blockDimY, unsigned int blockDimZ,
-        unsigned int sharedMemBytes, Stream hStream,
-        void** kernelParams)
-
-    # Kernel attributes
-    int cuFuncGetAttribute(int *pi, CUfunction_attribute attrib,
-                           Function hfunc)
-
-    int cuFuncSetAttribute(Function hfunc, CUfunction_attribute attrib,
-                           int value)
-
-    # Texture reference
-    int cuTexRefSetAddress(size_t* ByteOffset, TexRef hTexRef, Deviceptr dptr,
-                           size_t bytes)
-    int cuTexRefSetAddress2D(TexRef hTexRef, const Array_desc* desc,
-                             Deviceptr dptr, size_t Pitch)
-    int cuTexRefSetAddressMode(TexRef hTexRef, int dim, Address_mode am)
-    int cuTexRefSetArray(TexRef hTexRef, Array hArray, unsigned int Flags)
-    int cuTexRefSetBorderColor(TexRef hTexRef, float* pBorderColor)
-    int cuTexRefSetFilterMode(TexRef hTexRef, Filter_mode fm)
-    int cuTexRefSetFlags(TexRef hTexRef, unsigned int Flags)
-    int cuTexRefSetFormat(TexRef hTexRef, Array_format fmt,
-                          int NumPackedComponents)
-    int cuTexRefSetMaxAnisotropy(TexRef hTexRef, unsigned int maxAniso)
-
-    # Occupancy
-    int cuOccupancyMaxActiveBlocksPerMultiprocessor(
-        int* numBlocks, Function func, int blockSize, size_t dynamicSMemSize)
-    int cuOccupancyMaxPotentialBlockSize(
-        int* minGridSize, int* blockSize, Function func, CUoccupancyB2DSize
-        block2shmem, size_t dynamicSMemSize, int blockSizeLimit)
+IF CUPY_USE_CUDA_PYTHON:
+    from cuda.ccuda cimport *
+ELSE:
+    include '_driver_extern.pxi'
 
+cdef extern from '../../cupy_backend.h' nogil:
     # Build-time version
-    int CUDA_VERSION
     int HIP_VERSION
 
+# Provide access to constants from Python.
+from cupy_backends.cuda.api._driver_enum import *
```

Comment:
In the case of CUDA Python, can't we utilize the enums exposed to Python from `cudapython/cuda.pyx`? Though the current approach is probably a lot easier to manage.

Reply:
Thanks, applied!

Reply:
That's possible but we have to maintain `_driver_enum` for HIP anyways.